e2e/menu_items_inline/paragraphs.spec.js

```diff
@@ -0,0 +1,133 @@
+});
+
+test("should load paragraphs from the index file in the editor", async ({
+  page,
+}) => {
+  await expect(
+    page
+      .locator("#editor p")
+      .getByText("Hello world, this is the ProseMirror editor."),
+  ).toBeVisible();
+});
+
+test("should render typed paragraph text in the editor when clicking on 'p'", async ({
+  page,
+}) => {
+  await page.getByText("H2", { exact: true }).click();
+  await page.getByText("p", { exact: true }).click();
+
+  await page.locator("#editor .ProseMirror.govspeak").focus();
+  await page.keyboard.type("Testing!\n");
+
+  await expect(page.locator("#editor p").getByText("Testing!")).toHaveCSS(
+    "font-weight",
+    REGULAR_FONT_WEIGHT,
+  );
+});
+
+test.describe("with bold text", () => {
+  test("should load bold paragraphs from the index file in the editor", async ({
+    page,
+  }) => {
+    await expect(
+      page.locator("#editor p").getByText("This is some bold text."),
+    ).toHaveCSS("font-weight", BOLD_FONT_WEIGHT);
+  });
+
+  test("should render bold paragraph text in the editor and clear style after new line when clicking on 'B' and typing", async ({
+    page,
+  }) => {
+    await page.getByText("B", { exact: true }).click();
+    await page.locator("#editor .ProseMirror.govspeak").focus();
+    await page.keyboard.type("Testing bold!\nTesting not bold!\n");
+
+    await expect(
+      page.locator("#editor p").getByText("Testing bold!"),
+    ).toHaveCSS("font-weight", BOLD_FONT_WEIGHT);
+    await expect(
+      page.locator("#editor p").getByText("Testing not bold!"),
+    ).toHaveCSS("font-weight", REGULAR_FONT_WEIGHT);
+  });
+
+  test("should toggle bold paragraph text for the same line in the editor when clicking 'B' and typing", async ({
+    page,
+  }) => {
+    await page.getByText("B", { exact: true }).click();
+    await page.locator("#editor .ProseMirror.govspeak").focus();
+    await page.keyboard.type("Testing bold! ");
+
+    await page.getByText("B", { exact: true }).click();
+    await page.locator("#editor .ProseMirror.govspeak").focus();
+    await page.keyboard.type(" and not bold");
+
+    await expect(
+      page.locator("#editor p").getByText("Testing bold! and not bold"),
+    ).toBeVisible();
+    await expect(page.getByText("Testing bold!")).toHaveCSS(
+      "font-weight",
+      BOLD_FONT_WEIGHT,
+    );
+    await expect(page.getByText("and not bold")).toHaveCSS(
+      "font-weight",
+      REGULAR_FONT_WEIGHT,
+    );
+  });
+
+  test("should toggle bold paragraph text when selecting text and clicking 'B'", async ({
+    page,
+  }) => {
+    await page
+      .locator("#editor p")
+      .getByText("This is some bold text.")
+      .selectText();
+    await page.getByText("B", { exact: true }).click();
+
+    await expect(
+      page.locator("#editor p").getByText("This is some bold text."),
+    ).toHaveCSS("font-weight", REGULAR_FONT_WEIGHT);
+
+    await page
+      .locator("#editor p")
+      .getByText("This is some bold text.")
```

Comment:
Would there be value in testing a sub string here? I'm thinking that would prove that we've operating using a mark rather than a block

Reply:
I'm not quite sure what you mean ? this is the toggling test when you select some bold text, unbold it, and then bold it again. is it about the text selection rather than just clicking? cause for bold it doesn't affect the line nor the block - it only affects selected text.

Reply:
I suppose what I'm suggesting is making it clear that it's only going to affect selected text by testing a substring, rather than the whole line/block.

Reply:
ahhhhh ok got it - will make the change

Reply:
@ryanb-gds turns out highlighting substrings using playwright is a right pain... it only selects the entire element that contains the substring... considering the value vs effort i think maybe skip out on this one?